framework/include/problems/EigenProblem.h

```diff
@@ -84,7 +89,7 @@ class EigenProblem : public FEProblemBase
    * We pre-scale eigen vector using the same factor as the one computed in
    * "postScaleEigenVector"
    */
-  void preScaleEigenVector();
+  void preScaleEigenVector(std::pair<Real, Real> eig);
```

Comment:
```suggestion
  void preScaleEigenVector(const std::pair<Real, Real> & eig);
```

Reply:
donot think this is necessary because passing value for small size data types is ok, but I will do.

Reply:
I prefer to make things const wherever possible in order to express intent

Reply:
If passing value, it does not matter. I know there are arguments on this ;-)

Reply:
From a compiler standpoint it doesn't matter. But from an intent standpoint if that's const I can see that "whoever developed this API expects me to use this value and nothing else". const makes code more readable as it expresses intent